### PR TITLE
Redux: Added config to redux development middlewares that checks for mutations and serializability

### PR DESCRIPTION
**Pull request by torkelo**
A recent redux toolkit & redux upgrade merged this morning broke dashboards and explore (Strange that it passed e2e tests). 

This upgrade added stricter development middleware config that throws errors when detecting actions or states that contain non serializable state, as well as detection for state mutations. 

We store quite a bit of non seralizable things in state (for what I think are mostly good reasons), things like rxjs query subscriptions, data source instance, dashboard, panel models, plugin models. They all have some mutation or nonserializable parts to them. Think some can be refactored to be immutable in the future but that will take time. 

This adds a bunch of config to ignore some actions and state paths from these checks (these checks only run in development)

### CODE REVIEW

File: public/app/store/configureStore.ts [hunk 1]
```diff
@@ -22,11 +21,17 @@ export function configureStore() {
     },
   });
 
-  const middleware = process.env.NODE_ENV !== 'production' ? [toggleLogActionsMiddleware, thunk, logger] : [thunk];
+  const middleware = process.env.NODE_ENV !== 'production' ? [toggleLogActionsMiddleware, logger] : [];
+
+  const reduxDefaultMiddlewars = getDefaultMiddleware<StoreState>({
```
Comment (Clarity-89) [inline public/app/store/configureStore.ts:26]: A small typo, should be `reduxDefaultMiddlewares`or just `reduxDefaultMiddleware`.

**Review by Clarity-89 — APPROVED** [review]

**Review by dprokop — APPROVED** [review]

File: public/app/store/configureStore.ts (hunk 1 above)
Comment (dprokop) [inline public/app/store/configureStore.ts:26]:
```suggestion
  const reduxDefaultMiddleware = getDefaultMiddleware<StoreState>({
```

File: public/app/store/configureStore.ts [hunk 2]
```diff
@@ -22,11 +21,17 @@ export function configureStore() {
...
+    thunk: true,
+    serializableCheck: false,
+    immutableCheck: false,
+  } as any);
 
   const store = reduxConfigureStore<StoreState>({
     reducer: createRootReducer(),
-    middleware: [...getDefaultMiddleware<StoreState>(), ...middleware] as [ThunkMiddleware<StoreState>],
+    middleware: [...reduxDefaultMiddlewars, ...middleware] as [ThunkMiddleware<StoreState>],
```
Comment (dprokop) [inline public/app/store/configureStore.ts:34]:
```suggestion
    middleware: [... reduxDefaultMiddleware, ...middleware] as [ThunkMiddleware<StoreState>],
```

**Comment by torkelo** [comment]
I am afk right now, can anyone if you fix the change suggestion and merge?

**Comment by dprokop** [comment]
@torkelo done, will merge soon

**Review by Estrax — APPROVED** [review]

**Pull request merged**